File: kbs_bbs/bbs2www/html/bbstop10.php
<?php
	/**
	 * $Id$ 
	 */
	require("funcs.php");
	if ($loginok !=1 )
		html_nologin();
	else
	{
		$top_file = get_bbsfile("/etc/posts/day");
		$fp = fopen($top_file, "r");
		if ($fp == FALSE) {
		        html_init("gb2312");
			html_error_quit("无法读取十大话题数据");
                }
                $modifytime=filemtime($top_file);
                if (cache_header("public",$modifytime,300))
                	return;
	html_init("gb2312");
?>
<body>
<center><?php echo $BBS_FULL_NAME; ?> -- 今日十大热门话题
<hr>
<table border="1" width="610">
<tr><td>名次</td><td>讨论区</td><td>标题</td><td>作者</td><td>人数</td></tr>
<?php
		fgets($fp, 256);
		fgets($fp, 256);
		for ($i = 0; $i < 10 && !feof($fp); $i++)
		{
			$one_line = fgets($fp, 256);
			if ($one_line == FALSE)
				break;
			$r_board = trim(substr($one_line, 41, 16));
			$r_num = trim(substr($one_line, 98, 4));
			$r_id = trim(substr($one_line, 118, 12));
			$one_line = fgets($fp, 256);
			if ($one_line == FALSE)
				break;
			$r_title = trim(substr($one_line, 27, 60));
?>
<tr>
<td>第 <?php echo $i+1; ?> 名</td>
<td><a href="/cgi-bin/bbs/bbsdoc?board=<?php echo $r_board; ?>"><?php echo $r_board; ?></a></td>
<td><a href="/cgi-bin/bbs/bbstfind?board=<?php echo $r_board; ?>&title=<?php echo urlencode($r_title); ?>"><?php echo $r_title; ?></a></td>
<td><a href="/cgi-bin/bbs/bbsqry?userid=<?php echo $r_id; ?>"><?php echo $r_id; ?></a></td>
<td><?php echo $r_num; ?></td>
</tr>
<?php
		}
		fclose($fp);
?>
</table></center>
<?php
		html_normal_quit();
	}
?>

File: kbs_bbs/bbs2www/src/bbstop10.c
/*#include "bbslib.h"*/
#include "bbslib.h"

int main() {
	FILE *fp;
	int n;
	char s1[256], s2[256], s3[256], s4[256],s5;
	char brd[256], id[256], title[256], num[100];
	init_all();
	printf("<center>%s -- 今日十大热门话题\n<hr>\n", BBSNAME);
	fp=fopen("etc/posts/day", "r");
	if(fp==0) http_fatal("can't read data");
	fgets(s1, 255, fp);
	fgets(s1, 255, fp);
	printf("<table border=\"1\" width=\"610\">\n");
	printf("<tr><td>名次<td>讨论区<td>标题<td>作者<td>人数\n");
	for(n=1; n<=10; n++)
       	{
		if(fgets(s1, 255, fp) == NULL)
		       	break;
		sscanf(s1+41, "%s", brd);
		sscanf(s1+120, "%s", id);
		sscanf(s1+99, "%s", num);
		if(fgets(s1, 255, fp) == NULL)
		       	break;
		strsncpy(title, s1+27, 60);
		printf("<tr><td>第 %d 名<td><a href=\"bbsdoc?board=%s\">%s</a><td><a href='bbstfind?board=%s&title=%s'>%42.42s</a><td><a href=bbsqry?userid=%s>%12s</a><td>%s\n",
   n, brd, brd, brd, nohtml(title), nohtml(title), id, id, num);
	}
		printf("</table><center>");
		http_quit();
}
